src/test/java/htsjdk/samtools/cram/encoding/external/ExternalCodecEquivalenceTest.java

```diff
@@ -0,0 +1,224 @@
+package htsjdk.samtools.cram.encoding.external;
+
+import htsjdk.HtsjdkTest;
+import htsjdk.samtools.cram.encoding.CRAMCodec;
+import htsjdk.samtools.cram.io.IOTestCases;
+import htsjdk.samtools.util.RuntimeIOException;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExternalCodecEquivalenceTest extends HtsjdkTest {
+    // show that External codecs are equivalent within certain ranges
+
+    // if all values are 0 <= v <= 0x7F (high bit 7)
+    // ExternalByteCodec, ExternalIntegerCodec, and ExternalLongCodec streams are identical
+    // because raw bytes, ITF8 and LTF8 are identical
+
+    @Test(dataProvider = "testPositiveByteLists", dataProviderClass = IOTestCases.class)
+    public void byteEquivalenceTest(final List<Byte> values) {
+        codecPairTestForByte(values, this::writeByteForByte, this::readIntegerForByte);
+        codecPairTestForByte(values, this::writeByteForByte, this::readLongForByte);
+        codecPairTestForByte(values, this::writeIntegerForByte, this::readByteForByte);
+        codecPairTestForByte(values, this::writeIntegerForByte, this::readLongForByte);
+        codecPairTestForByte(values, this::writeLongForByte, this::readByteForByte);
+        codecPairTestForByte(values, this::writeLongForByte, this::readIntegerForByte);
+    }
+
+    private void codecPairTestForByte(final List<Byte> values, ByteWriter writer, ByteReader reader) {
+        byte[] written = writer.write(values);
+        final List<Byte> read = reader.read(written, values.size());
+        Assert.assertEquals(read, values);
+    }
+
+    // if all values are 0 <= v <= 0x0F FF FF FF (high bit 28)
+    // ExternalIntegerCodec, and ExternalLongCodec streams are identical
+    // because ITF8 and LTF8 are identical
+
+    @Test(dataProvider = "testUint28Lists", dataProviderClass = IOTestCases.class)
+    public void intEquivalenceTest(final List<Integer> values) {
+        codecPairTestForInteger(values, this::writeIntegerForInteger, this::readLongForInteger);
+        codecPairTestForInteger(values, this::writeLongForInteger, this::readIntegerForInteger);
+    }
+
+    private void codecPairTestForInteger(final List<Integer> values, IntegerWriter writer, IntegerReader reader) {
+        byte[] written = writer.write(values);
+        final List<Integer> read = reader.read(written, values.size());
+        Assert.assertEquals(read, values);
+    }
+
+    private interface ByteWriter {
+        byte[] write(final List<Byte> toWrite);
+    }
+
+    private interface ByteReader {
+        List<Byte> read(final byte[] toRead, final int count);
+    }
+
+    private byte[] writeByteForByte(List<Byte> values) {
```

Comment:
It would be super helpful if each of these methods that uses a particular codec had a name that reflected/included the codec its using.